### PR TITLE
Add support for minifying index.html

### PR DESCRIPTION
**Pull request by ishitatsuyuki**
This was built with compatibility in mind.

### CODE REVIEW

**Comment by ishitatsuyuki** [comment]
Travis spurious error. Please trigger a rebuild.

**Comment by dave11mj** [comment]
@ishitatsuyuki you can trigger a rebuilt by `git push -f` it also gives you the chance to rebase your branch so its up to date with `master` ~ ^^

**Comment by ishitatsuyuki** [comment]
Two weeks elapsed. Please take a look!

**Comment by filipesilva** [comment]
Heya @ishitatsuyuki, old system are used with the CLI, and we really don't want to introduce breaking changes for such little benefit. I'm sorry, but I'll have to close this PR.

**Comment by ishitatsuyuki** [comment]
@filipesilva I'm aware of such concerns. Actually, I'm willing to disable the removal of trailing slash (it can be specified as `[keepTrailingSlash](https://github.com/kangax/html-minifier#options-quick-reference)`, if it's going to keep compatibility with (really) old XHTML engines.

It's a tradeoff of compatibility with performance, though. If you reopened this, I will change it to allow XHTML ASAP.

**Comment by filipesilva** [comment]
I don't quite get what the problem is that you're trying to solve though. Is there a bug currently that you're trying to fix?

**Comment by ishitatsuyuki** [comment]
#1861

**Comment by filipesilva** [comment]
Ah I see now! I'll reopen, and thank you for taking such an old issue on. 

I'm slightly worried that it might affect some users that are currently depending on the current `index.html` format, but that's a matter for discussion. We're also thinking of adding a `--minify` flag so this could be controlled there for less disruption.

As for this PR as is, can you change it to not remove `xhtml`, and thus not change the other files? Your changes to `packages/@angular/cli/tasks/eject.ts` are just whitespace changes as well, and we normally don't allow that in PRs (it's just back and forth on file that don't need to change).

**Comment by ishitatsuyuki** [comment]
Looks like there's some presumably spurious build failures. Please trigger a rebuild and see if there is a real problem or not.

**Comment by clydin** [comment]
Some performance data if anyone is curious.

File size differences for a new project:

Minified | Raw (bytes) | Gzip (bytes)
--- | --- | ---
No | 701 | 377
Yes | 674 | 365
Change | -27 (~4%) | -12 (~3%)

For a medium-size project with some additional index content:

Minified | Raw (bytes) | Gzip (bytes)
--- | --- | ---
No | 1163  | 550
Yes | 1121 | 532
Change | -42 (~4%) | -18 (~3%)

File: packages/@angular/cli/models/webpack-configs/browser.ts [hunk 1]
```diff
@@ -37,7 +37,13 @@ export function getBrowserConfig(wco: WebpackConfigOptions) {
         filename: path.resolve(buildOptions.outputPath, appConfig.index),
         chunksSortMode: packageChunkSort(appConfig),
         excludeChunks: lazyChunks,
-        xhtml: true
+        xhtml: true,
+        cache: false,
```
Comment (filipesilva) [inline packages/@angular/cli/models/webpack-configs/browser.ts]: Why is cache set to false?

**Comment by filipesilva** [comment]
Restarted builds.

File: packages/@angular/cli/models/webpack-configs/browser.ts (hunk 1 above)
Comment (ishitatsuyuki) [inline packages/@angular/cli/models/webpack-configs/browser.ts]: Looks like a leftover from the inlining implementation. Will remove it.

**Comment by clydin** [comment]
might want to adjust the commit message to indicate this just minifies `index.html`.

**Comment by ishitatsuyuki** [comment]
Review?

**Comment by clydin** [comment]
It looks like [version 3.0](https://github.com/jantimon/html-webpack-plugin/issues/558) of the `HtmlWebpackPlugin` will remove built-in support for minification.  Not sure on the timeline though.

**Comment by ishitatsuyuki** [comment]
Semver will still protect us, and I believe someone will create a plugin to bring the functionality. If nobody do, maybe I will do that.

:( so slow review, it's going to enter the refactor cycle

**Comment by filipesilva** [comment]
Sorry for the delay in getting it merged, I'm doing it now.

**Comment by icorne** [comment]
Is there any way we can still get the trailing slash? Or to disable the minification of just the index.html?

**Pull request merged**